There's another bug in RewriteMap handling in Apache 1.3.17, which
causes ${} expansions to be completely ignored. This patch fixes it.

Index: src/modules/standard/mod_rewrite.c
===================================================================
RCS file: /home/cvs/apache-1.3/src/modules/standard/mod_rewrite.c,v
retrieving revision 1.169
retrieving revision 1.170
diff -u -r1.169 -r1.170
--- src/modules/standard/mod_rewrite.c	2001/01/18 22:28:34	1.169
+++ src/modules/standard/mod_rewrite.c	2001/02/01 04:12:26	1.170
@@ -2287,11 +2287,11 @@
 		char *map, *key, *dflt, *result;
 		char xkey[MAX_STRING_LEN];
 		char xdflt[MAX_STRING_LEN];
-		key = find_char_in_brackets(inp, ':', '{', '}');
+		key = find_char_in_brackets(inp+2, ':', '{', '}');
 		if (key == NULL)
 		    goto skip;
 		map  = ap_pstrndup(r->pool, inp+2, key-inp-2);
-		dflt = find_char_in_brackets(inp, '|', '{', '}');
+		dflt = find_char_in_brackets(key+1, '|', '{', '}');
 		if (dflt == NULL) {
 		    key  = ap_pstrndup(r->pool, key+1, endp-key-1);
 		    dflt = "";
